internal/nearcache/record_store.go

```diff
@@ -394,3 +566,75 @@ func (rs *RecordStore) updateRecordValue(rec *Record, value interface{}) error {
 	rec.SetValue(value)
 	return nil
 }
+
+func (rs *RecordStore) onExpire() {
+	// port of: com.hazelcast.internal.nearcache.impl.store.AbstractNearCacheRecordStore#onExpire
+	rs.incrementExpirations()
+}
+
+func getEvictionPolicyComparator(cfg *nearcache.EvictionConfig) nearcache.EvictionPolicyComparator {
+	cmp := cfg.Comparator()
+	if cmp != nil {
+		return cmp
+	}
+	switch cfg.EvictionPolicy() {
+	case nearcache.EvictionPolicyLRU:
+		return LRUEvictionPolicyComparator
+	case nearcache.EvictionPolicyLFU:
+		return LFUEvictionPolicyComparator
+	case nearcache.EvictionPolicyRandom:
+		return RandomEvictionPolicyComparator
+	case nearcache.EvictionPolicyNone:
+		return nil
+	}
+	msg := fmt.Sprintf("unknown eviction polcy: %d", cfg.EvictionPolicy())
+	panic(ihzerrors.NewIllegalArgumentError(msg, nil))
+}
+
+type evictionCandidate struct {
+	key       interface{}
+	evictable *Record
+}
+
+func (e evictionCandidate) Key() interface{} {
+	return e.key
+}
+
+func (e evictionCandidate) Value() interface{} {
+	return e.evictable.Value()
+}
+
+func (e evictionCandidate) Hits() int64 {
+	return int64(e.evictable.Hits())
+}
+
+func (e evictionCandidate) CreationTime() int64 {
+	return e.evictable.CreationTime()
+}
+
+func (e evictionCandidate) LastAccessTime() int64 {
+	return e.evictable.LastAccessTime()
+}
+
+func evaluateForEviction(cmp nearcache.EvictionPolicyComparator, candies []evictionCandidate) evictionCandidate {
+	now := time.Now().UnixMilli()
+	var selected evictionCandidate
+	var hasSelected bool
+	for _, current := range candies {
+		// initialize selected by setting it to current candidate.
+		if !hasSelected {
+			selected = current
```

Comment:
Should you set hasSelected here as well? Also the name candies looks strange :D It means "şekerler"

Reply:
No, the code above is port of com.hazelcast.internal.eviction.impl.evaluator.EvictionPolicyEvaluator#evaluate I'll add a comment

Reply:
Added at: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/0d49f98e13b8fc0d4843db1366d353fb6a96a174

Reply:
Explanation: So, you are using hasSelected instead of the `null` value. So when selected is assigned to a nonnull value in Java code, you should set hasSelected to `true`.  I am assuming `current` variable in `for (C current : candidates) {` is not `null` though. As far as I understand, `sample()` method does not return `null`.

Reply:
You guys are right, here's the update: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/24624ef616401d25ca45b672f2c963f2ff050f55